### PR TITLE
Fix TODO to use int32 when parsing JSON in paho iot hub twin sample.

### PR DESCRIPTION
**Pull request by ahsonkhan**
cc @momuno, @danewalton

### CODE REVIEW

**Comment by momuno** [comment]
I don't think there is any need for a temp_property_value anymore.  (Previously it was cast to an int32_t)

Can this line change to:
AZ_RETURN_IF_FAILED(az_json_token_get_int32(&json_parser.token, &reported_property_value));
?

File: sdk/samples/iot/hub/src/paho_iot_hub_twin_example.c [hunk 1]
```diff
@@ -558,15 +558,14 @@ static az_result update_property(az_span desired_payload)
...
-      AZ_RETURN_IF_FAILED(az_json_token_get_uint32(&json_parser.token, &temp_property_value));
+      int32_t temp_property_value = 0;
+      AZ_RETURN_IF_FAILED(az_json_token_get_int32(&json_parser.token, &temp_property_value));
 
       printf(
           "Updating \"%.*s\" locally.\n",
           az_span_size(reported_property_name),
           az_span_ptr(reported_property_name));
-      reported_property_value = (int32_t)temp_property_value;
+      reported_property_value = temp_property_value;
```
Comment (momuno) [inline sdk/samples/iot/hub/src/paho_iot_hub_twin_example.c:568]: not needed now.

File: sdk/samples/iot/hub/src/paho_iot_hub_twin_example.c [hunk 2]
```diff
@@ -558,15 +558,14 @@ static az_result update_property(az_span desired_payload)
       // Move to the value token
       AZ_RETURN_IF_FAILED(az_json_parser_next_token(&json_parser));
 
-      // TODO: Change back to int32_t once that is supported.
-      uint32_t temp_property_value = 0;
-      AZ_RETURN_IF_FAILED(az_json_token_get_uint32(&json_parser.token, &temp_property_value));
+      int32_t temp_property_value = 0;
```
Comment (momuno) [inline sdk/samples/iot/hub/src/paho_iot_hub_twin_example.c:561]: temp_property_value not needed

File: sdk/samples/iot/hub/src/paho_iot_hub_twin_example.c [hunk 3]
```diff
@@ -558,15 +558,14 @@ static az_result update_property(az_span desired_payload)
...
+      AZ_RETURN_IF_FAILED(az_json_token_get_int32(&json_parser.token, &temp_property_value));
```
Comment (momuno) [inline sdk/samples/iot/hub/src/paho_iot_hub_twin_example.c:562]: change to
AZ_RETURN_IF_FAILED(az_json_token_get_int32(&json_parser.token, &reported_property_value));

?

File: sdk/samples/iot/hub/src/paho_iot_hub_twin_example.c (hunk 2 above)
Comment (danewalton) [reply sdk/samples/iot/hub/src/paho_iot_hub_twin_example.c:561]: yea i don't think this intermediary step is needed

File: sdk/samples/iot/hub/src/paho_iot_hub_twin_example.c (hunk 3 above)
Comment (ahsonkhan) [reply sdk/samples/iot/hub/src/paho_iot_hub_twin_example.c:562]: What was the intention of having the `temp_property_value` in the first place?

Comment (momuno) [reply sdk/samples/iot/hub/src/paho_iot_hub_twin_example.c:562]: Casting to int32_t.  Had to initially work with an uint32_t.  Prior to that I think we were using double and then casting to int.

**Pull request merged**